### PR TITLE
Led viewer branch merge

### PR DESCRIPTION
**Pull request by davepl**
Adds the LED Color Data server to provide frames back to clients that want to render the current state of the matrix.

* [X] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [X] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [X] I selected `main` as the target branch.
* [X] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

File: include/ledviewer.h [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    int                         _port;
+    int                         _server_fd;
+    struct sockaddr_in          _address; 
+
+    const int BUFFER_SIZE = 1024;
+
+public:
+
+    size_t              _cbReceived;
```
Comment (rbergen) [inline include/ledviewer.h]: I was wondering what purpose this variable serves, as it doesn't seem to be used anywhere? Also, why is it public? 
(Both questions actually also apply to audio.cpp, but that's technically out of the scope of this PR.)

File: src/audio.cpp [hunk 2]
```diff
@@ -110,15 +110,7 @@ void IRAM_ATTR AudioSamplerTaskEntry(void *)
 #include <fcntl.h>
 
 /** Returns true on success, or false if there was an error */
```
Comment (rbergen) [inline src/audio.cpp]: I think this comment line can be removed as well?

**Review by rbergen — APPROVED** [review]
Couple of very minor points/questions, but LGTM!

**Comment by davepl** [comment]
Hmm…. I think that’s copy-paste floatsam from the class I started with!  It’s unneeded and should be removed.

- Dave

**Comment by rbergen** [comment]
Done. I'll merge when CI completes. I assume I can start working on the merge conflicts with PlummersSoftwareLLC:main after?

**Comment by davepl** [comment]
I've been pulling main into davepl regularly, so hopefully there aren't many/any big conflicts!

**Comment by rbergen** [comment]
There weren't; there were just a few minor things to sync. With those now sorted, I think davepl:main is ready to be merged into PlummersSoftwareLLC:main.

**Pull request merged**